### PR TITLE
Fix(DB/Creature&Gameobject) fix issue 4085

### PR DESCRIPTION
**Pull request by r0m1ntik**
## Changes Proposed:
-  Remove event_scripts for gameobject (status)
-  Add smart_scripts for gameobject status
-  Add script for npc and text 

## Issues Addressed:
- Closes [4085](https://github.com/azerothcore/azerothcore-wotlk/issues/4085)

## Tests Performed:
-
-

## How to Test the Changes:

1) .go ob id 181964
2) .q a 9683
3) use object
4) enjoy

## Target Branch(es):
- [x] Master

 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1609431870979648000.sql [hunk 1]
```diff
@@ -0,0 +1,19 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1609431870979648000');
+UPDATE `gameobject_template` SET `Data2` = 0, `AIName` = 'SmartGameObjectAI' WHERE `entry` = 181964;
+DELETE FROM `event_scripts` WHERE `id` = 11027;
+DELETE FROM `smart_scripts` WHERE `entryorguid` IN (181964, 17715, 1771500);
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/rev_1609431870979648000.sql:4]: you need to specify the `source_type` as well when using `DELETE` on `smart_scripts`

File: data/sql/updates/pending_db_world/rev_1609431870979648000.sql [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+INSERT INTO `smart_scripts` VALUES
+	(181964, 1, 0, 0, 70, 0, 100, 0, 2, 0, 0, 0, 0, 12, 17715, 4, 300000, 0, 0, 0, 8, 0, 0, 0, 0, -1896.13, -12861.1, 87.1412, 3.5412, 'Statue of Queen Azshara - On Gameobject State Change - Summon Creature \'Atoph the Bloodcursed\'');
+INSERT INTO `smart_scripts` VALUES
+	(17715, 0, 0, 0, 63, 0, 100, 0, 0, 0, 0, 0, 0, 80, 1771500, 2, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Atoph the Bloodcursed - Just Summoned - Run Script');
+INSERT INTO `smart_scripts` VALUES
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/rev_1609431870979648000.sql:9]: you can just have one `INSERT INTO `smart_scripts` VALUES`, no need to have 3. See: https://www.azerothcore.org/wiki/Dealing-with-SQL-files#write-compact-code

**Review by FrancescoBorzi — CHANGES_REQUESTED** [review]

File: data/sql/updates/pending_db_world/rev_1609431870979648000.sql (hunk 1 above)
Comment (r0m1ntik) [reply data/sql/updates/pending_db_world/rev_1609431870979648000.sql]: done

File: data/sql/updates/pending_db_world/rev_1609431870979648000.sql (hunk 2 above)
Comment (r0m1ntik) [reply data/sql/updates/pending_db_world/rev_1609431870979648000.sql:5]: done

**Review by Kitzunu — COMMENTED** [review]
remvoe all indents pls

**Comment by FrancescoBorzi** [comment]
@r0m1ntik still WIP ?

**Comment by Helias** [comment]
it is not in WIP anymore

File: data/sql/updates/pending_db_world/rev_1609431870979648000.sql [hunk 3]
```diff
@@ -0,0 +1,17 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1609431870979648000');
+UPDATE `gameobject_template` SET `Data2` = 0, `AIName` = 'SmartGameObjectAI' WHERE `entry` = 181964;
+DELETE FROM `event_scripts` WHERE `id` = 11027;
+DELETE FROM `smart_scripts` WHERE `entryorguid` IN (181964, 17715, 1771500) AND `source_type` IN (0, 1, 9);
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/rev_1609431870979648000.sql:4]: the `DELETE` condition is wrong, it should be instead something like:

```sql
DELETE FROM `smart_scripts` WHERE 
(`entryorguid` = 181964 AND `source_type`= 1) OR
(`entryorguid` = 17715 AND `source_type` = 0) OR
(`entryorguid` = 1771500 AND `source_type` = 9);
```

**Review by FrancescoBorzi — CHANGES_REQUESTED** [review]

**Comment by FrancescoBorzi** [comment]
@r0m1ntik any update about this? have you checked my comments? also have you tested it in game?

**Comment by ChromieCraftServer** [comment]
any update on this?

**Comment by claudiodfc** [comment]
https://github.com/azerothcore/azerothcore-wotlk/pull/6477 Fixes both issues opened on the quest. You can close this PR.

**Pull request closed without merging**